### DISCUSSION TITLE
How to use have multiple cache with useQueries but make single api call ?

### DISCUSSION THREAD

**Discussion by MrRobz** [TanStack/query#4577] [Q&A]
We have an api that can send the results of multiple posts at a time.

GET api/posts?id=1,2

Please help me with some optimisation here.
I want to be able to store each post in its separate cache. Using  `useQueries` for that. But the problem is it makes multiple API calls.

```
 const results = useQueries([
   { queryKey: ['post', 1], queryFn: fetchPost },
   { queryKey: ['post', 2], queryFn: fetchPost },
 ])
```

How do update this so that **fetchPost is only called once** and can make API call fetching both the posts `api/posts?id=1,2`?

**Comment by TkDodo** [comment]
Have a look at the discussion about batching:
- https://github.com/TanStack/query/discussions/365

the accepted answer shows a really good way how to do this with `dataloader`
